userspace/libsinsp/sinsp_filtercheck_fspath.cpp

```diff
@@ -349,7 +348,14 @@ uint8_t* sinsp_filter_check_fspath::extract(sinsp_evt* evt, OUT uint32_t* len, b
 			return NULL;
 		}
 
-		m_tstr = sinsp_utils::concatenate_paths(tinfo->get_cwd(), m_tstr);
+		if(m_tstr.front() != '/')
+		{
+			m_tstr = sinsp_utils::concatenate_paths(tinfo->get_cwd(), m_tstr.c_str());
+		} else
+		{
+			// concatenate_paths takes care of resolving the path
+			m_tstr = sinsp_utils::concatenate_paths("", m_tstr.c_str());
```

Comment:
`std::string` should be automatically cast to `std::string_view`, I don't think you need the `c_str()` call (which needs an automatic cast from `const char*` to `std::string_view` anyway). A similar simplification can be done a few lines above.

Reply:
We had that discussion before, nothing else worked here ...

Reply:
Please, forgive me, I forgot. I'm aging. I'm noticing that we do pass a straight `std::string` to this function elsewhere. Do you mean that in this specific call instance it is not working? As in not compiling or not doing the right concatenation?

Reply:
It's not doing the right concatenation. I developed and tested on my fedora Linux box.